keyboards/keyboardio/atreus/keymaps/zosiu/keymap.c

```diff
@@ -0,0 +1,265 @@
+#define KC_S_WD LT(_MOD_S, KC_S)  // press for s, hold for dia_s layer
+#define KC_Z_WD LT(_MOD_Z, KC_Z)  // press for z, hold for dia_z layer
+
+#define KC_COMS M(_KC_COMS)       // comma + space
+#define KC_ENOP ALT_T(KC_ENT)     // press for enter, hold for option
+#define KC_BSCM GUI_T(KC_BSPC)    // press for backspace, hold for command
+#define KC_CTES CTL_T(KC_ESC)     // press for escape, hold for control
+#define KC_SHTB SFT_T(KC_TAB)     // press for tab, hold for shift
+#define KC_SPFN LT(_FUN, KC_SPC)  // press for space, hold for function layer (aka spacefn)
+#define KC_CMTB RGUI(KC_TAB)      // command + tab
+#define KC_OPTB RALT(KC_TAB)      // optoin + tab
+#define KC_CSTB S(RCTL(KC_TAB))   // shift + control + tab
+#define KC_C_TB RCTL(KC_TAB)      // control + tab
+#define KC_C_LF RCTL(KC_LEFT)     // control + left
+#define KC_C_RT RCTL(KC_RGHT)     // control + right
+#define KC_BRUP KC_F15            // brightness up on osx
+#define KC_BRDN KC_F14            // brightness down on osx
+
+enum unicode_names {
+    _HU_AA_U,
+    _HU_AA_L,
+    _HU_II_U,
+    _HU_II_L,
+    _HU_UU_U,
+    _HU_UU_L,
+    _HU_EE_U,
+    _HU_EE_L,
+    _HU_OO_U,
+    _HU_OO_L,
+    _HU_OE_U,
+    _HU_OE_L,
+    _HU_UE_U,
+    _HU_UE_L,
+    _HU_OEE_U,
+    _HU_OEE_L,
+    _HU_UEE_U,
+    _HU_UEE_L,
+
+    _PL_AU_U,
+    _PL_AU_L,
+    _PL_EU_U,
+    _PL_EU_L,
+    _PL_LE_U,
+    _PL_LE_L,
+    _PL_CI_U,
+    _PL_CI_L,
+    _PL_NI_U,
+    _PL_NI_L,
+    _PL_SI_U,
+    _PL_SI_L,
+    _PL_ZI_U,
+    _PL_ZI_L,
+    _PL_ZE_U,
+    _PL_ZE_L
+};
+
+const uint32_t PROGMEM unicode_map[] = {
+    [_HU_AA_L]  = 0x00E1,  //á
+    [_HU_AA_U]  = 0x00C1,  //Á
+    [_HU_II_L]  = 0x00ED,  //í
+    [_HU_II_U]  = 0x00CD,  //Í
+    [_HU_UU_L]  = 0x00FA,  //ú
+    [_HU_UU_U]  = 0x00DA,  //Ú
+    [_HU_EE_L]  = 0x00E9,  //é
+    [_HU_EE_U]  = 0x00C9,  //É
+    [_HU_OO_L]  = 0x00F3,  //ó
+    [_HU_OO_U]  = 0x00D3,  //Ó
+    [_HU_OE_L]  = 0x00F6,  //ö
+    [_HU_OE_U]  = 0x00D6,  //Ö
+    [_HU_UE_L]  = 0x00FC,  //ü
+    [_HU_UE_U]  = 0x00DC,  //Ü
+    [_HU_OEE_L] = 0x0151,  //ő
+    [_HU_OEE_U] = 0x0150,  //Ő
+    [_HU_UEE_L] = 0x0171,  //ű
+    [_HU_UEE_U] = 0x0170,  //Ű
+
+    [_PL_AU_L] = 0x0105,  //ą
+    [_PL_AU_U] = 0x0104,  //Á
+    [_PL_EU_L] = 0x0119,  //ę
+    [_PL_EU_U] = 0x0118,  //Ę
+    [_PL_LE_L] = 0x0142,  //ł
+    [_PL_LE_U] = 0x0141,  //Ł
+    [_PL_CI_L] = 0x0107,  //ć
+    [_PL_CI_U] = 0x0106,  //Ć
+    [_PL_NI_L] = 0x0144,  //ń
+    [_PL_NI_U] = 0x0143,  //Ń
+    [_PL_SI_L] = 0x015B,  //ś
+    [_PL_SI_U] = 0x015A,  //Ś
+    [_PL_ZI_L] = 0x017A,  //ź
+    [_PL_ZI_U] = 0x0179,  //Ź
+    [_PL_ZE_L] = 0x017C,  //ź
+    [_PL_ZE_U] = 0x017B   //Ż
+};
+
+void eeconfig_init_user(void) { set_unicode_input_mode(UC_LNX); }
+
+void persistant_default_layer_set(uint16_t default_layer) {
+    eeconfig_update_default_layer(default_layer);
+    default_layer_set(default_layer);
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt) {
```

Comment:
`action_get_macro` has been deprecated, and needs to be migrated to <https://docs.qmk.fm/#/feature_macros?id=the-new-way-send_string-amp-process_record_user>

Reply:
I migrated one part of it. The activation/deactivation of the oneshot layer however wouldn't work in `process_record_user` no matter how I tried, so I left that as is for now.

Reply:
Try using `post_process_record_user` and see if that works better. 

If that works, then I think I know what the issue is, then. :(